### DISCUSSION TITLE
ValidationError: Entity of type Permission expected for property Role.permissions

### DISCUSSION THREAD

**Discussion by imcm7** [mikro-orm/mikro-orm#4706] [General]
Can I use only ids for `user.roles.assign([1,2,3]);`?
Attach and detach like here https://docs.adonisjs.com/guides/models/relationships#attach

**Comment by B4nan** [comment]
`em.assign(user, { roles: [1, 2, 3] })`

**Reply by B4nan** [reply]
You did it wrong, look closer on what I told you. Also dont forget to await the flush call.
